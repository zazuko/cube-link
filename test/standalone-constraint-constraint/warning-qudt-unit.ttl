PREFIX unit: <http://qudt.org/vocab/unit/>
PREFIX qudt: <http://qudt.org/schema/qudt/>
@prefix rdf: <http://www.w3.org/1999/02/22-rdf-syntax-ns#> .
@prefix cube: <https://cube.link/> .
@prefix observation: <https://environment.ld.admin.ch/foen/nfi/observation/max_min_undefined> .
@prefix sh: <http://www.w3.org/ns/shacl#> .
@prefix xsd: <http://www.w3.org/2001/XMLSchema#> .
@prefix schema: <http://schema.org/> .
@base <https://example.org/> .

<cube> a cube:Cube ;
    cube:observationConstraint <shape> ;
    cube:observationSet <observationSet> .

<observationSet> cube:observation <observationA>, <observationB>, <observationC> .

<observationA> a cube:Observation ;
    cube:observedBy <observer> ;
    <dimension1> 4.9 .

<observationB> a cube:Observation ;
    cube:observedBy <observer> ;
    <dimension1> 1.0 .

<observationC> a cube:Observation ;
    cube:observedBy <observer> ;
    <dimension1> ""^^cube:Undefined .

<shape> a cube:Constraint ;
    sh:targetClass cube:Observation ;
    sh:closed true ;
    sh:property
        [
            sh:path rdf:type ;
            sh:nodeKind sh:IRI ;
            sh:minCount 1 ;
            sh:maxCount 1
        ] ;
    sh:property
        [
            sh:path cube:observedBy ; ;
            sh:nodeKind sh:IRI ;
            sh:minCount 1 ;
            sh:maxCount 1
        ] ;
    sh:property
        [
            sh:datatype xsd:decimal ;
            sh:path <dimension1> ;
            schema:name "some Quantity" ;
            sh:minCount 1 ;
            sh:maxCount 1 ;
            qudt:unit unit:KiloGM ;
        ] ;
    sh:property
        [
            sh:datatype xsd:decimal ;
            sh:path <dimension2> ;
            schema:name "some Quantity" ;
            sh:minCount 1 ;
            sh:maxCount 1 ;
            qudt:hasUnit unit:KiloGM ;
        ] ;
.
