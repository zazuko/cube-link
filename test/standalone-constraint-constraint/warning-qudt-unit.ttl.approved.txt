@prefix sh: <http://www.w3.org/ns/shacl#> .
@prefix rdf: <http://www.w3.org/1999/02/22-rdf-syntax-ns#> .
@prefix schema: <http://schema.org/> .
@prefix xsd: <http://www.w3.org/2001/XMLSchema#> .
@prefix cube: <https://cube.link/> .

_:report a sh:ValidationReport ;
	sh:result [
		rdf:type sh:ValidationResult ;
		sh:resultSeverity sh:Warning ;
		sh:sourceConstraintComponent sh:OrConstraintComponent ;
		sh:sourceShape [
			sh:path sh:property ;
			sh:or (
				[
					sh:path <http://qudt.org/schema/qudt/unit> ;
					sh:equals <http://qudt.org/schema/qudt/hasUnit> ;
				]
				[
					sh:path <http://qudt.org/schema/qudt/unit> ;
					sh:maxCount 0 ;
				]
			) ;
			sh:message "The use of qudt:unit is deprecated, please use qudt:hasUnit instead" ;
			sh:severity sh:Warning ;
		] ;
		sh:focusNode <https://example.org/shape> ;
		sh:resultPath sh:property ;
		sh:value _:b3 ;
		sh:resultMessage "The use of qudt:unit is deprecated, please use qudt:hasUnit instead" ;
	] ;
	sh:conforms false .
